master/buildbot/reporters/bitbucketserver.py

```diff
@@ -125,6 +129,201 @@ def send(self, build):
                     ))
 
 
+class BitbucketServerCoreAPIStatusPush(http.HttpStatusPushBase):
+    name = "BitbucketServerCoreAPIStatusPush"
+    secrets = ["token", "auth"]
+    neededDetails = dict(wantProperties=True)
+
+    def checkConfig(self, base_url, token=None, auth=None, **kwargs):
+        if not base_url:
+            config.error("Parameter base_url has to be given")
+        if token is not None and auth is not None:
+            config.error("Only one authentication method can be given "
+                "(token or auth)")
+        super().checkConfig(**kwargs)
+
+    @defer.inlineCallbacks
+    def reconfigService(self, base_url, token=None, auth=None,
+                        name=None, nameSuffix=None, startDescription=None,
+                        endDescription=None, key=None, parent=None,
+                        buildNumber=None, ref=None, duration=None,
+                        testResults=None, verbose=False, debug=None,
+                        verify=None, **kwargs):
+        yield super().reconfigService(**kwargs)
+        self._name = name
+        self.nameSuffix = nameSuffix
+        self.startDescription = startDescription or 'Build started.'
```

Comment:
Buildbot has recently changed the preferred code style to use PEP8 where possible (i.e. everywhere except public, documented API). See [here](http://docs.buildbot.net/latest/developer/style.html).

Could we use `snake_case` for variables and private members, where possible?

Reply:
```suggestion
        self.start_description = startDescription or 'Build started.'
```
```suggestion
        self.start_description = start_description or 'Build started.'
```
Just to be sure, is the first suggestion the right one or is it the second one? As the parameters are probably considered public.

Reply:
First one is preferred. Indeed, parameters are part of public interface and we don't change them just yet.